projects/hipfft/clients/samples/hipfft_multigpu_2d_z2z.cpp

```diff
@@ -32,9 +33,57 @@ DISABLE_WARNING_RETURN_TYPE
 #include <hip/hip_runtime_api.h>
 DISABLE_WARNING_POP
 
+void sneaky(std::vector<std::complex<double>> &cinput,
+            const int Nx, const int Ny, const int direction)
+{
+    hipError_t           hip_rt;
+    hipfftDoubleComplex* x;
+
+    using ArrayType = typename std::remove_reference<decltype(cinput)>::type;
+    using ValueType = typename std::remove_extent<ArrayType>::type;
+    size_t complex_bytes = sizeof(ValueType) * cinput.size();
+    
+    //size_t complex_bytes = sizeof(std::remove_extent<decltype(std::remove_reference<cinput>)::value_type>) * cinput.size();
+
+    
+    hip_rt = hipMalloc(&x, complex_bytes);
+    if(hip_rt != hipSuccess)
+        throw std::runtime_error("hipMalloc failed");
+    hip_rt = hipMemcpy(x, cinput.data(), complex_bytes, hipMemcpyHostToDevice);
+    if(hip_rt != hipSuccess)
+        throw std::runtime_error("hipMemcpy failed");
+    hipfftHandle plan{};
+    hipfftResult hipfft_rt = hipfftCreate(&plan);
+    if(hipfft_rt != HIPFFT_SUCCESS)
+        throw std::runtime_error("hipfftCreate failed");
+
+    hipfft_rt = hipfftPlan2d(&plan, // plan handle
+                             Nx, // transform length
+                             Ny, // transform length
+                             HIPFFT_Z2Z); // transform type (HIPFFT_C2C for single-precision)
```

Comment:
```suggestion
    hipfftResult hipfft_rt = hipfftPlan2d(&plan, // plan handle
                             Nx, // transform length
                             Ny, // transform length
                             HIPFFT_Z2Z); // transform type (HIPFFT_C2C for single-precision)
```
`hipfftPlan2d` does re-create a plan and overwrite the previously-created one otherwise (leak).

Alternatively, we could keep `hipfftCreate` and use `hipfftMakePlan2d` instead of `hipfftPlan2d`.